packages/schema-generator/src/generate_schema.ts

```diff
@@ -126,3 +132,24 @@ export const parseDatabaseUrl = (databaseUrl: string): SQLDataSourceConfig => {
     );
   }
 };
+
+const constructDBEngine = (engine: string): SQLEngine => {
+  switch (engine) {
+    case 'mysql':
+      return 'mysql';
+    case 'postgresql':
+    case 'postgres':
+      return 'postgresql';
+    default:
+      throw new Error(`Unsupported database engine: ${engine}`);
+  }
+};
+
+const getDefaultPort = (engine: SQLEngine): number => {
+  switch (engine) {
+    case 'mysql':
+      return 3306;
+    case 'postgresql':
+      return 5432;
+  }
+};
```

Comment:
Should this logic go to `@aws-amplify/graphql-schema-generator` ? I.e. make port optional in `TypescriptDataSchemaGeneratorConfig` and just pass through from this package ?

Reply:
I would like to keep this as part of `parseDatabaseUrl` helper. The way we have it now is this method will extract all the required attributes and pass it to TypescriptSchemaGenerator.

Possibly we move the `parseDatabaseUrl` to `@aws-amplify/graphql-schema-generator` but I would like to take care of it in a different PR.